src/pair.cpp

```diff
@@ -786,14 +786,14 @@ void Pair::ev_setup(int eflag, int vflag, int alloc)
   evflag = 1;
 
   eflag_either = eflag;
-  eflag_global = eflag % 2;
-  eflag_atom = eflag / 2;
+  eflag_global = eflag & ENERGY_GLOBAL;
+  eflag_atom = eflag & ENERGY_PER_ATOM;
```

Comment:
@rbberger This is creating the wrong value. Similar to what happened at #2469

```cpp
eflag=0 -> 0
eflag=1 -> 0
eflag=2 -> 2
eflag=3 -> 2
```
It should be something like `bool(i & ENERGY_PER_ATOM)`

Reply:
> @rbberger This is creating the wrong value. Similar to what happened at #2469

the value doesn't matter. all the tests are for zero vs. non-zero.

[...]

> It should be something like `bool(i & ENERGY_PER_ATOM)`

if you absolutely want 0 vs 1 it should be more like `(eflag & ENERGY_PER_ATOM) ? 1 : 0`
those flags are "int" not "bool".

Reply:
@akohlmey I fixed Kim to avoid the requirement for value. Just mentioned the value for the backward compatibility and if there is any other piece of code that checks the value otherwise you are right, the value is not important